detect.go

```diff
@@ -0,0 +1,40 @@
+package composer
+
+import (
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"os"
+	"path/filepath"
+)
+
+func Detect() packit.DetectFunc {
+	return func(context packit.DetectContext) (packit.DetectResult, error) {
```

Comment:
I was thinking that `BP_COMPOSER_VERSION` should be checked in `composer-install` and be part of the `Requirement` that buidpack generates.

I haven't seen a version on the `Provides`, I took a look at some other buildpacks such as `poetry`, which does check for existence of `pyproject.toml` in order to detect and does not include a version when it `Provides poetry.`

Reply:
One thing I'll note: if a buildpack provides `composer` and then optionally requests `composer` when `BP_COMPOSER_VERSION` is present, it means that using only this buildpack without `BP_COMPOSER_VERSION` fails detection but using only this buildpack with `BP_COMPOSER_VERSION` performs the install. 

I had anticipated that this buildpack would only install `composer` if it was requested by a subsequent buildpack, especially because it will always detect. So using this buildpack in any combination of buildpacks with `BP_COMPOSER_VERSION` set will always install `composer`.

Or maybe I'm thinking of this the wrong way? Do we anticipate using the buildpack always in some larger order group that undergoes detection as a whole?

Reply:
@joshuatcasey Yea, I definitely see your point. It is odd and unintuitive that the buildpack would pass detection on it's own if `BP_COMPOSER_VERSION` is set. Like above I'd like to hear from @thitch97 about the decisions made in the Poetry buildpacks.

The way I think about it is that with `BP_COMPOSER_VERSION` **not set**, its delegating to the subsequent buildpacks to figure out what versions to use. The composer buildpack needs to wait to see if any buildpack down the line needs a specific version of composer to do whatever they are doing. I think we're in agreement about that.

The reason for checking `BP_COMPOSER_VERSION` in the composer buildpack, is that it's the highest precedence option. If the user knows what they want up front, cut out the need for checking any other versions down the line and tell the composer buildpack ASAP what it should provide. Yes it can pass detection on it's own, but I can't see why anyone would find themselves in that use case unlesss they specifically only included composer dist buildpack in their build.

Reply:
Got it. I'm starting to track. I do wish that the the mechanism to "force install" Composer were more explicit. 

But I can see the idea that `BP_COMPOSER_VERSION` is used to force a particular version of Composer regardless of what else happens in the order group.